### DISCUSSION TITLE
Detect expanded state of drop down menu

### DISCUSSION THREAD

**Discussion by mattjpoole** [MyreMylar/pygame_gui#509] [Q&A] [answered]
HI,

I have drop down menu whose expanded state overlaps rectangles which are clickable as part of a game. Beause the collidepoint of the game rect's and mouse still intersect when the drop down is selected the rectangles react underneath the expanded drop down.

Is there a way of detecting that the drop down has expanded (and closed) so I can enable/disable the game rects collidepoint check when the drop down is in use?

Many Thanks in advance
Matt

**Comment by MyreMylar** [comment, marked as answer]
Hey matt,

You should be able to use the return value of `ui_manager.process_events()` to see if pygame GUI used a click (or other event). Failing that the drop down itself has:

```
if(your_dropdown.current_state == your_dropdown.menu_states['expanded']):
    # dropdown is expanded
``` 

Hope that helps.

**Reply by mattjpoole** [reply]
Hey,

Thank you for your reply. 

I tried looking at other events. A button click is dispactched by pygame gui when you open and close it. However, its also possible to close the drop down by clicking anywhere and that doesnt dispatch the button click event, so my game is stuck in that edge case. 

I'll try your suggestion and confirm whether it works when I have done so.

Many Thanks
Matt

**Comment by mattjpoole** [comment]
Hi,

Thanks it worked. I went with:

` def dropdown_is_expanded(self) -> bool :
        return type(self.level_drop_down.current_state) == pygame_gui.elements.ui_drop_down_menu.UIExpandedDropDownState`

Just trying to avoid the string literal.

Thanks again for your help
Matt
